smithy-model/src/test/resources/software/amazon/smithy/model/transform/dedicated-input-output/handles-dangling-single-reference-before.smithy

```diff
@@ -0,0 +1,13 @@
+$version: "2.0"
+
+namespace smithy.example
+
+// We have a shape being used by both input and output
+// so there will be an extra or "dangling" reference
```

Comment:
nit: This comment is a little confusing without the context of the PR and original issue. Maybe it can just describe what this test is showing, that a shared input/output shape gets cleaned up and replaced by the appropriate input and output shapes.